### ISSUE TITLE
Memory usage keeps increasing when using tf.node.decodeImage

### ISSUE THREAD

**Issue by Agrafador** [tensorflow/tfjs#6380]
**System information**
@tensorflow/tfjs@3.16.0
Using CPU on Windows 10 and Linux Ubuntu 20.04.4

**Bug description**
I'm tracing a memory leak related to a model prediction task of images, and the conclusion that i end up with is that the method `tf.node.decodeImage` is the origin of the memory leak.

I already tried multiple things including `startScope` & `endScope`, but the RSS memory keeps rising while heap remains stable.
If you use the code below you will see that the RSS memory keeps rising even though that dispose is called. This increase of memory is much larger if you use more images.
`tf.memory` also prints everything as empty.

**Code to replicate the issue**

```
const tf = require('@tensorflow/tfjs-node');
const axios = require('axios');

const downloadFile = async (url) => {
    const req = await axios({
        url,
        responseType: 'arraybuffer',
    });
    return req.data;
};

const main = async () => {
    const imageBuffer = await downloadFile(
        'https://e3.365dm.com/22/05/1600x900/skynews-rice-paddies-rambukkana_5763666.jpg?bypass-service-worker',
    );

    for (let i = 0; i < 900; i += 1) {
        const decodedImage = tf.node.decodeImage(imageBuffer, 3);
        tf.dispose([decodedImage]);
        console.log(`Iteration ${i}`);
        console.log(tf.memory());

        const used = process.memoryUsage();
        for (const key in used) {
            console.log(`${key} ${Math.round((used[key] / 1024 / 1024) * 100) / 100} MB`);
        }
    }
};
main();
```

**Output**
```
Iteration 0
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 92.22 MB
heapTotal 38.25 MB
heapUsed 23.94 MB
external 1.86 MB
arrayBuffers 0.4 MB

(...)

Iteration 899
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 244.16 MB
heapTotal 40.5 MB
heapUsed 25.07 MB
external 1.81 MB
arrayBuffers 0.35 MB
```

**Comment by pyu10055** [comment]
@mattsoulanille is this related to the decodeJpeg mem leak? Seems here it is not using string for the file name.

**Comment by rthadur** [comment]
@Agrafador have you tried using latest version 3.16 ?

**Comment by Agrafador** [comment]
> @Agrafador have you tried using latest version 3.16 ?

Yes, it was the version that i used initially

**Comment by mattsoulanille** [comment]
Thanks, @Agrafador, for the report! I believe this is a duplicate of #5242, and is fixed in 3.17.0 by #6371. I ran an extended version of your test on 3.17.0, and while initially, memory seems to climb, it levels out after ~7000 iterations.
```
Iteration 0
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 152.93 MB
heapTotal 37.44 MB
heapUsed 26.62 MB
external 1.36 MB
arrayBuffers 0.63 MB

Iteration 1000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 201.4 MB
heapTotal 38.69 MB
heapUsed 28.4 MB
external 1.36 MB
arrayBuffers 0.19 MB

Iteration 2000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 205.25 MB
heapTotal 39.69 MB
heapUsed 29.02 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 3000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 206.34 MB
heapTotal 39.69 MB
heapUsed 28.1 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 4000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 206.45 MB
heapTotal 39.69 MB
heapUsed 26.53 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 5000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.16 MB
heapTotal 39.69 MB
heapUsed 24.69 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 6000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.16 MB
heapTotal 39.69 MB
heapUsed 30.49 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 7000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 28.47 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 8000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 26.38 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 9000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 24.3 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 10000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 30.02 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 11000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 27.93 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 12000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 25.79 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 13000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 23.77 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 14000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 29.48 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 15000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 222.23 MB
heapTotal 39.69 MB
heapUsed 27.37 MB
external 0.92 MB
arrayBuffers 0.19 MB
```
This matches the behavior of 3.3.0.

In 3.16.0, memory grows rapidly (above 1Gb by iteration 5000):
```
Iteration 0
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 155.18 MB
heapTotal 37.76 MB
heapUsed 26.86 MB
external 1.36 MB
arrayBuffers 0.63 MB

Iteration 1000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 377.72 MB
heapTotal 39.01 MB
heapUsed 28.57 MB
external 1.36 MB
arrayBuffers 0.19 MB

Iteration 2000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 523.39 MB
heapTotal 40.51 MB
heapUsed 29.03 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 3000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 719.38 MB
heapTotal 40.51 MB
heapUsed 27.9 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 4000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 877.32 MB
heapTotal 40.51 MB
heapUsed 26.44 MB
external 0.92 MB
arrayBuffers 0.19 MB

Iteration 5000
{ unreliable: true, numTensors: 0, numDataBuffers: 0, numBytes: 0 }
rss 1030.5 MB
heapTotal 40.51 MB
heapUsed 24.51 MB
external 0.92 MB
arrayBuffers 0.19 MB

^C
```

**Comment by mattsoulanille** [comment]
Closing as fixed by #6371.
